parlai/core/torch_agent.py

```diff
@@ -402,6 +465,51 @@ def get_history_vec_list(self):
         """
         return self.history_vecs
 
+    def get_history_str_with_timesteps(self) -> Optional[str]:
+        """
+        Return the string version of the history, including timesteps.
+        """
+        if len(self.history_strings) > 0:
+            history = self.history_strings[:]
+            timesteps = self.history_timesteps[:]
+            history = self.delimiter.join(
+                [f"{h}{self.delimiter}{t}" for h, t in zip(history, timesteps)]
```

Comment:
could we have a separate flag for the timestep delimiter? since we may not want to separate by newline

Reply:
I've added a new flag `--timestep-delimiter` that defaults to `\t`.
For line 492 I've kept `delimiter` though, since I want `history_timestep_vec` to match `timestep_vec`